### PR TITLE
feat(store): enable immutability checks by default

### LINKED ISSUE #2217: Store: enable immutability checks by default
- [ ] Enable the immutability runtime checks for the StoreModule config.
- [ ] Update/remove the console warning message
- [ ] Update schematics

### PR DESCRIPTION
**Pull request by evgenyfedorenko**
Closes #2217

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?
Store runtime immutability checks are being enabled by default with this change

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently immutability checks are opt in.

Closes #2217

## What is the new behavior?
If state or action is mutated then there will be a run time exception thrown by default now. In order to opt out user will need to pass immutability checks set to false.

## Does this PR introduce a breaking change?

```
[x] Yes
[] No
```

BREAKING CHANGE

Before
Immutability checks are opt-in.

After
Immutability checks are enabled by default. If state or action is mutated then there will be a run time exception thrown.

### CODE REVIEW

**Comment by evgenyfedorenko** [comment]
Please let me know if there is anything else I need to add/correct.

File: modules/store/schematics/ng-add/index.spec.ts [hunk 1]
```diff
@@ -131,14 +131,16 @@ describe('Store ng-add Schematic', () => {
     expect(content).toMatch(/export interface AppState {/);
   });
 
-  it('should add runtime checks by default', () => {
-    const options = { ...defaultOptions, module: 'app.module.ts' };
+  it('should not add runtime checks by default', () => {
+    const options = {
+      ...defaultOptions,
+      module: 'app.module.ts',
+      minimal: true,
```
Comment (timdeschryver) [inline modules/store/schematics/ng-add/index.spec.ts]: Why is this added?

Comment (evgenyfedorenko) [reply modules/store/schematics/ng-add/index.spec.ts]: to check if schematics creating the correct StoreModule.forRoot statement?
Maybe rename the test so it makes more sense ?

Comment (timdeschryver) [reply modules/store/schematics/ng-add/index.spec.ts]: I was pointing at the `minimal` flag, because it wasn't there before.
But I think this test can be removed, it was added to verify the immutability flags were added.

**Comment by timdeschryver** [comment]
@evgenyfedorenko could you also add a breaking change note to the PR please, you can find a template in [CONTRIBUTING.MD](https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#footer)

**Comment by evgenyfedorenko** [comment]
> @evgenyfedorenko could you also add a breaking change note to the PR please, you can find a template in [CONTRIBUTING.MD](https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#footer)

Done
